src/statements.md

```diff
@@ -54,17 +54,36 @@ fn outer() {
 > **<sup>Syntax</sup>**\
 > _LetStatement_ :\
 > &nbsp;&nbsp; [_OuterAttribute_]<sup>\*</sup> `let` [_PatternNoTopAlt_]
->     ( `:` [_Type_] )<sup>?</sup> (`=` [_Expression_] )<sup>?</sup> `;`
+>     ( `:` [_Type_] )<sup>?</sup> ( | `=` [_Expression_] | `=` [_Expression_]
+>     `else` [_BlockExpression_]) `;`
 
-A *`let` statement* introduces a new set of [variables], given by an
-irrefutable [pattern]. The pattern is followed optionally by a type
-annotation and then optionally by an initializer expression. When no
-type annotation is given, the compiler will infer the type, or signal
+A *`let` statement* introduces a new set of [variables], given by a [pattern].
+The pattern is followed optionally by a type annotation and then either ends,
+or is followed by an initializer expression plus an optional `else` block.
+When no type annotation is given, the compiler will infer the type, or signal
 an error if insufficient type information is available for definite
 inference. Any variables introduced by a variable declaration are visible
 from the point of declaration until the end of the enclosing block scope,
 except when they are shadowed by another variable declaration.
 
+If an `else` block is not present, the pattern has to be irrefutable.
+If an `else` block is present, the pattern may be refutable, and if it is, it
+will be checked on run time. On a mismatch, the `else` block is executed.
+The `else` block has to always diverge (evaluate to type `!`).
+If an `else` block is present, restrictions apply on the expression:
+It might not be a [_LazyBooleanExpression_], or end in a `}` token.
```

Comment:
I think this should be up in the grammar section since it is part of the grammar.

Also, similar to the must/should/may terminology:

```suggestion
If an `else` block is present, restrictions apply on the expression:
It must not be a [_LazyBooleanExpression_], or end in a `}` token.
```

Although I do lean towards the earlier wording I suggested. I think it can sometimes be a little difficult to follow when using "it" when it is not very clear what "it" is. Here it is referring to an "expression", but there are different expressions at play (such as the block expression, and all the expressions within the block). Up in the grammar, the word _Expression_ refers to a specific expression (there is only one of that form; that is a rule with a capital E and italicized).

Reply:
I think it's clear because there is only one expression in the grammar. It is also mentioned in the first paragraph.